### ISSUE TITLE
Cop idea: Dangerous Method Suffix

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12339]
## Description

The names of potentially _dangerous_ methods (i.e. methods that modify `self` or the arguments, `exit!` (doesn’t run the finalizers like `exit` does), etc) should end with an exclamation mark if there exists a safe version of that dangerous method.

https://rubystyle.guide/#dangerous-method-bang

## Behavior

```ruby
# bad - there is no matching 'safe' method
class Person
  def update!
  end
end

# good
class Person
  def update
  end
end

# good
class Person
  def update!
  end

  def update
  end
end
```

**Comment by fatkodima** [comment]
If this will be implemented, it should be disabled by default. It can generate many false positives because of the mixed in modules with the methods.

**Comment by koic** [comment]
This cannot be detected mechanically.
